### DISCUSSION TITLE
nvchad setup

### DISCUSSION THREAD

**Discussion by membranobruno** [chipsenkbeil/distant#244] [Q&A]
Hi! I'm using nvchad  and trying to setup the distant. The distant itself is installed and I can run command in terminal. I have added 

```
  {
    "chipsenkbeil/distant.nvim",
    branch = "v0.3",
    config = function()
      require("distant"):setup()
    end,
  },
```

to plugins.lua and restarted nvim but have no result. 
Not an editor command: DistantInstall

thanks for any help

**Comment by chipsenkbeil** [comment]
What happens if you run the setup command manually after loading up neovim?

```
:lua require("distant"):setup()
```

Do you get any error? Does the command `:DistantInstall` show up? Also, if you already have `distant` available in the commandline and it's the required version (at the moment, `0.20.0`), then you shouldn't need to run `:DistantInstall` at all.

**Comment by BBScholar** [comment]
NvChad lazy loads plugins by default. There are a couple solutions here. 
1. Disable lazy loading for distant. This will (mildly) impact startup time if you care about that.

```lua
 {
    "chipsenkbeil/distant.nvim",
    branch = "v0.3",
    lazy = false,
    config = function()
      require("distant"):setup()
    end,
  },
```

2. Add commands that will lazy load distant when called.
```lua
 {
    "chipsenkbeil/distant.nvim",
    branch = "v0.3",
    cmd = {
         "DistantLaunch",
         "DistantConnect",
          .... 
    },
    config = function()
      require("distant"):setup()
    end,
  },
```
I prefer solution 2, but you'll need to type all the commands out.
